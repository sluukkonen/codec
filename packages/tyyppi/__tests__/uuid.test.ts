import * as crypto from "node:crypto"
import { uuid } from "../src/index.js"
import { expectParseFailure, expectParseSuccess } from "./helpers.js"

test("should parse valid v4 UUIDs", () => {
  expectParseSuccess(uuid, crypto.randomUUID())
})

test("should parse NIL uuid", () => {
  expectParseSuccess(uuid, "00000000-0000-0000-0000-000000000000")
})

test("should not parse other strings", () => {
  expectParseFailure(uuid, "asdf")
})

test("should not parse non-strings", () => {
  expectParseFailure(uuid, null)
})
