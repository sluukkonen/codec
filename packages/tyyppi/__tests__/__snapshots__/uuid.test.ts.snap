// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`should not parse non-strings 1`] = `
[
  {
    "code": "invalid_type",
    "expected": "string",
    "message": "Expected a string, received null",
    "path": [],
    "received": "null",
  },
]
`;

exports[`should not parse other strings 1`] = `
[
  {
    "code": "invalid_uuid",
    "message": "Expected the string to be an UUID",
    "path": [],
  },
]
`;
